src/input.rs

```diff
@@ -466,17 +494,20 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
     }
 
     pub fn mouse_input(&mut self, state: ElementState, button: MouseButton, modifiers: ModifiersState) {
-        if let MouseButton::Left = button {
-            let state = mem::replace(&mut self.ctx.mouse_mut().left_button_state, state);
-            if self.ctx.mouse_mut().left_button_state != state {
-                match self.ctx.mouse_mut().left_button_state {
-                    ElementState::Pressed => {
-                        self.on_mouse_press(modifiers);
-                    },
-                    ElementState::Released => {
-                        self.on_mouse_release(modifiers);
-                    }
-                }
+        let button_state = match button {
```

Comment:
Might be good to change this to `old_state` (or something like that) to clearly indicate that this is the result of the `mem::replace` and it is neither the current `_button_state` nor the current `state`.

Reply:
Agreed. Updated to `prev_state` to match other uses of `mem::replace` in the file.